packages/angular_devkit/build_angular/src/angular-cli-files/utilities/find-specs.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Path, getSystemPath, join } from '@angular-devkit/core';
+import * as glob from 'glob';
+
+export function findSpecs(input: string, workspaceRoot: Path, sourceRoot: string): string[] {
+
+  input = input.replace(/\\/g, '/'); // normalize input
+
+  // remove source root to support relative paths from root
+  // such paths are easy to get when running script via IDEs
+  if (sourceRoot && input.startsWith(sourceRoot + '/')) {
+    input = input.substr(sourceRoot.length + 1); // +1 to include slash
+  }
+
+  if (input.endsWith('.ts') && !input.endsWith('.spec.ts')) {
```

Comment:
I'm not sure I agree with this bit. Some projects might have their own spec suffixes that we shouldn't assume.

Reply:
I think I'll remove the option and require user to pass explicitly the file they want to run. 

To me this is useful sometimes - I can modify ts file, run custom vscode command to run selected test (while ts file is selected) and matching unit test ends up running.

Reply:
You know, I was thinking about this a bit more and I think it's ok to assume `.spec.ts`, as long as that does not cause valid filenames to be ignored. We don't really give users options to change the spec suffix when generating files, but we don't stop them from changing it afterwards (and it will still work, given they change `test.ts` as well).

So for instance, if the file is `something.custom.ts`, and there does indeed exist a file on disk called `something.custom.spec.ts`, it's ok to use that one instead. But if `something.custom.spec.ts` doesn't exist, the original file should still be used.

WDYT? Does this make sense with the usecases you already have at your company?

Reply:
I am perfectly OK without assuming too many things. Such cases/conventions would need to be documented as well, otherwise it might be just me using them :)